apps/mobile/src/components/FilterOffersScreen/components/LocationSearch/LocationsList.tsx

```diff
@@ -0,0 +1,44 @@
+import {Stack} from 'tamagui'
+import {FlatList} from 'react-native'
+import {type LocationSuggestion} from '@vexl-next/rest-api/dist/services/location/contracts'
+import LocationCell from './LocationCell'
+import {type Atom} from 'jotai'
+import atomKeyExtractor from '../../../../utils/atomUtils/atomKeyExtractor'
+
+interface Props {
+  locationSuggestionsAtoms: Array<Atom<LocationSuggestion>>
+  onClose: () => void
+}
+
+function renderItem(
+  locationSuggestionAtom: Atom<LocationSuggestion>,
+  onClose: () => void
+): JSX.Element {
+  return (
+    <LocationCell
+      locationSuggestionAtom={locationSuggestionAtom}
+      onPress={onClose}
+    />
+  )
+}
+
+function LocationsList({
+  locationSuggestionsAtoms,
+  onClose,
+}: Props): JSX.Element {
+  return (
+    <Stack pt={'$2'}>
+      <FlatList
+        showsHorizontalScrollIndicator={false}
+        showsVerticalScrollIndicator={false}
+        keyExtractor={atomKeyExtractor}
+        data={locationSuggestionsAtoms}
+        renderItem={({item: locationSuggestionAtom}) =>
```

Comment:
This neglects the purpouse of defining the `renderItem` function outside of the component. We need the props to stay equal between renders, but here, `renderItem` receives different function on every render. 

In this case, it might be a good idea to use `useMemo` . 

See [this](https://stackoverflow.com/questions/74333190/react-native-whole-flatlist-rerenders-on-every-change-even-with-react-memo) for further explanation on why this is a good idea to do.